### DISCUSSION TITLE
Event handlers for workflow status changes?

### DISCUSSION THREAD

**Discussion by bjornw77** [Netflix/conductor#2316] [Q&A]
Hi! Is there any way to register some sort of event handler for workflow status changes?

I have a Java microservice with a CRUD Rest API, and requests to each endpoint starts a new workflow instance, like this:

`final String workFlowInstanceId = workflowClient.startWorkflow(createVehicleProfileRequest);`

Once the workflow is completed, I want to return an aggregated result (of all tasks defined in the workflow) to the client that invoked my Rest API. The WorkflowClient API has a getWorkflow method, like this:

`workflowClient.getWorkflow(workFlowInstanceId, false)`

But I assume there must be a way for the Conductor server to push status changes of workflows to it's clients, rather than the client having to poll the server and check for the same?

**Comment by apanicker-nflx** [comment]
Conductor server does not push status chages to client, rather the server provides an interface with a couple of [different implementations](https://github.com/Netflix/conductor/tree/main/contribs/src/main/java/com/netflix/conductor/contribs/listener) for publishing/acting upon workflow termination/completion events. You could implement your own listener to route these events to your clients.
